### ISSUE TITLE
mitmweb relative path support doesn't still support flows related requests

### ISSUE THREAD

**Issue by ariel42** [mitmproxy/mitmproxy#6525]
#### Problem Description
Hi,
Thank you for your great project!

In this pull request - https://github.com/mitmproxy/mitmproxy/pull/6411 - relative paths support was introduced, which is great, but still it has few bugs related to flows requests: For example, when running the web interface under http://127.0.0.1/mitmweb the "Save" menu item to download the flows doesn't work, as it redirects the browser to /flows/dump, which is a root relative path again.
Searching with chrome DevTools for "/flows" with the leading slash reveals more potential occurrences for this bug.

#### Steps to reproduce the behavior:
1. Run mitmweb under a sub path
2. Capture some traffic
3. Try to save it with the "Save" menu item

#### System Information
10.1.5

**Comment by lups2000** [comment]
@mhils is this still an open issue?

**Comment by davet2001** [comment]
I think it is not solved yet. If someone is able to help me with #6562 then I'm happy to finish it. Alternatively, they could create a new PR starting from where I got to.

**Comment by lups2000** [comment]
> I think it is not solved yet. If someone is able to help me with #6562 then I'm happy to finish it. Alternatively, they could create a new PR starting from where I got to.

Yes, I can try to help! :) First, may I ask how you can start mitmweb under a subpath? Second, was your problem mainly due to failing tests?

**Comment by davet2001** [comment]
@lups2000 Yes, the main issue was the tests. I'd be grateful for your help!

The reason I was running under a subpath was by adding mitmproxy as a home assistant addon:
https://github.com/davet2001/home-assistant-addons

**Comment by lups2000** [comment]
> @lups2000 Yes, the main issue was the tests. I'd be grateful for your help!
> 
> The reason I was running under a subpath was by adding mitmproxy as a home assistant addon: https://github.com/davet2001/home-assistant-addons

okay!  Could you please tell me how to run mitmweb under a subpath in order to test the issue/your code? thanks :)

**Comment by davet2001** [comment]
@lups2000 I was using a home assistant addon which creates a proxy operating at a sub path, then forwards everything to a docker container.

It been a little while since I've worked on it so I can't remember precise details of how to get it running, but effectively it is an addon development environment for home assistant. 

This creates multiple docker containers which talk to each other.

The front end web server at the base url is in one container and the mitmweb server is running in another.

I used vscode.
